### ISSUE TITLE
Support Bikram-Sambat-calendar (Vikram-Samvat)

### ISSUE THREAD

**Issue by MenoData** [MenoData/Time4J#849]
This is the official calendar in Nepal. See also:

- http://nepalicalendar.rat32.com/
- https://en.wikipedia.org/wiki/Vikram_Samvat
- http://www.kultur-in-asien.de/Kalender/seite206.htm

A tabular-based solution is possible but only exists with limited data in the gregorian year range 1934-2023. An astronomical solution based on old hindu calendar as published in "Calendrical calculations" (Dershowitz/Reingold) should be evaluated, too.

This solar calendar needs to be distinguished from the lunar calendar called "Nepal Sambat". Latter one is used for the calculation of festivals/holidays.

**Comment by pragyanone** [comment]
I have always struggled to fully use open source apps (like [GnuCash](https://github.com/Gnucash/gnucash), [Debitum](https://github.com/Marmo/debitum)) because of this issue. [Bikram Sambat](https://en.wikipedia.org/wiki/Vikram_Samvat) calendar is not supported yet by operating softwares (Windows, Android, etc), and it is unnatural for app developers to code a calendar separately.

I have a python script for date conversion from (and to) this system and the ubiquitous [Gregorian](https://www.wikiwand.com/en/Gregorian_calendar) calendar. [(the python script)](https://github.com/pragyanone/adbs)

And I lack the coding knowledge to contribute to the apps by adding this snippet of code.

**Comment by MenoData** [comment]
That seems to be great news to hear. A wider range "adStart, adEnd = "1918-4-13", "2039-4-12" would be nice. May I ask you what source do you get the data mapping from?

Anyway, I will try to find some time to implement the calendar in October or November. Before, I am unfortunately too busy with other things. Thank you for your contribution.

**Comment by pragyanone** [comment]
My source is `https://pujann.com.np/ad-bs-converter.php`.

I don't understand this project,`Time4j`. Is it a module that other developers can import in their apps (like _GnuCash_) ?

**Comment by MenoData** [comment]
Yes, Time4J is a Java library project which can be imported by Java apps (for example using Maven). The sister project Time4A is for Android apps.
